src/utilities/mergeParams.ts

```diff
@@ -0,0 +1,15 @@
+import { Param } from '@/types'
+
+export function mergeParams(...params: Record<string, Param[]>[]): Record<string, Param[]> {
+  return params.reduce<Record<string, Param[]>>((combined, collection) => {
+    Object.entries(collection).forEach(([key, params]) => {
+      if (key in combined) {
+        params.forEach(param => combined[key].push(param))
+      } else {
+        combined[key] = params
```

Comment:
We might want to spread into a new array here. Otherwise we'll be update this first tuple in place with any additional values that get merged into it.

Reply:
so so? 

```ts
combined[key] = [...params]
```